### PR TITLE
Add demos for in-class React exercises

### LINKED ISSUE #72: Add demos to show goal of exercises in React module
**Which module(s) and week(s) does this change effect?**
Module(s): React
Week(s): 1 - 3

**What is the work that needs to be done?**

Add a demo for each each exercise in the React module, which demonstrates the final goal of the exercise.

**Why is this work important to do?**

The exercises are written to break the task down step-by-step, to help students work through the exercise. However this leads to confusion over what is required from the task. Additionally advanced students don't get the opportunity to try the exercise without the assistance.

**Additional context**

This is a common bit of feedback from teachers running the React module.

**Who might need to know about this change?**

@nbogie

### PR DESCRIPTION
**Pull request by alismith-dev**
## What does this change?

Module: React
Week(s): 1- 3

## Description

Closes https://github.com/CodeYourFuture/syllabus/issues/72

Pretty much does what it says on the tin. Adds components to demonstrate the final state of (relevant) Pokedex in-class exercises.

Example screenshot:

Screenshot 2020-10-17 at 15 21 37

To view the demos themselves, you'll probably have to pull and run locally as rendering React is not supported by Github's markdown parser.

## Who needs to know about this?

@ChrisOwen101 
@nbogie

### CODE REVIEW

File: docs/react/week-1/lesson.md [hunk 1]
```diff
@@ -5,6 +5,9 @@ sidebar_label: Lesson
 ---
 
 import Feedback from "@theme/Feedback";
+import { ExerciseDDemo } from './demos/ExerciseD.js'
```
Comment (ChrisOwen101) [inline docs/react/week-1/lesson.md:11]: Could we add a comment here explaining how this works and linking away to other documentation?

**Comment by ChrisOwen101** [comment]
Really really nice. 🎉

Could we add some links to the Instructor Notes to the source code of the Demos so teachers can reference them for solutions?

**Review by ChrisOwen101 — APPROVED** [review]
Reviewed and looks excellent.

File: docs/react/week-1/lesson.md (hunk 1 above)
Comment (alismith-dev) [reply docs/react/week-1/lesson.md:11]: Good point, added in 98106f0

**Comment by 40thieves** [comment]
> Could we add some links to the Instructor Notes to the source code of the Demos

Good idea, added in ff40124

**Review by ChrisOwen101 — APPROVED** [review]
Perfect - good to go!

**Comment by 40thieves** [comment]
Seems my merge failed the build https://github.com/CodeYourFuture/syllabus/runs/1298066768?check_suite_focus=true, Rerunning to see if it's a 1-off. Any ideas @ChrisOwen101?

**Comment by 40thieves** [comment]
Hmm, failing on `npm run build` locally too.

**Comment by 40thieves** [comment]
Nvm, @ChrisOwen101 says he broke the build yesterday

**Pull request merged**